src/azul/project/hca/metadata_generator.py

```diff
@@ -101,36 +97,22 @@ def _flatten(self, master, obj, parent):
             else:
                 self._set_value(master, newkey, value)
 
-    def _get_schema_name_from_object(self, object):
+    def _get_schema_name_from_object(self, object: JSON):
         if "describedBy" in object:
             return object["describedBy"].rsplit('/', 1)[-1]
         raise MissingDescribedByError("found a metadata without a describedBy property")
 
-    def _cmp_keys(self, a, b):
-        best_a = sys.maxsize
-        best_b = sys.maxsize
-
-        for idx, val in enumerate(self.default_order):
-            p = re.compile(val)
-
-            match_a = p.match(a)
-            match_b = p.match(b)
-
-            if match_a:
-                if idx < best_a:
-                    best_a = idx
-
-            if match_b:
-                if idx < best_b:
-                    best_b = idx
-
-        if best_a == best_b:
-            if a >= b:
-                return 1
-            return -1
-        return best_a - best_b
-
-    def add_bundle_files_to_row(self, bundle_uuid, bundle_version, list_of_metadata_objects, dir_name=None):
+    def add_bundle_files_to_row(self,
+                                bundle_uuid: str,
+                                bundle_version: str,
+                                list_of_metadata_objects: List) -> None:
+        """
+        :param bundle_uuid: The uuid of the bundle being digested
+        :param bundle_version: The version of the bundle being digested
+        :param list_of_metadata_objects: A list of metadata objects associated with this bundle
+
+        This provisions the required metadata to process the bundle files into metadata rows
```

Comment:
Just remove the docstring. It only adds to the confusion. Focus on adequate variable names and correct type hints.